src/API/Google/Middleware.php

```diff
@@ -649,4 +663,44 @@ public function is_subaccount(): bool {
 		// since transients don't support booleans, we save them as 0/1 and do the conversion here
 		return boolval( $is_subaccount );
 	}
+
+	/**
+	 * Performs a request to Google Shopping Data Integration (SDI) to get required information in order to form an auth URL.
+	 *
+	 * @return array An array with the JSON response from the WCS server.
+	 * @throws NotFoundExceptionInterface  When the container was not found.
+	 * @throws ContainerExceptionInterface When an error happens while retrieving the container.
```

Comment:
It can also throw an Exception: "Error Authenticating Google App..."

Reply:
Fixed here thx. https://github.com/woocommerce/google-listings-and-ads/pull/2424/commits/5b9a462e304af83a9332114bd12271eb1c4896a4